### PR TITLE
Docs: Add Explore guide for plugins

### PR DESCRIPTION
**Pull request by marcusolsson**
**What this PR does / why we need it**:

This PR adds a guide for how to add Explore support to a data source plugin.

### CODE REVIEW

File: docs/sources/developers/plugins/add-support-for-explore-queries.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
++++
+title = "Add support for Explore queries"
+type = "docs"
++++
+
+# Add support for Explore queries
+
+This guide explains how to add support for [Explore]({{< relref "../../features/explore.md" >}}) to an existing data source.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/add-support-for-explore-queries.md:8]: The last part of this sentence confuses me. Do you mean data source plugin?

File: docs/sources/developers/plugins/add-support-for-explore-queries.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+This guide assumes that you're already familiar with how to [Build a data source plugin]({{< relref "../../../../../tutorials/build-a-data-source-plugin.md" >}}).
+
+By adding support for Explore to your data source, users can make ad-hoc queries without the new of a dashboard. This is useful when users want to troubleshoot or to learn more about the data.
+
+To support Explore, your plugin needs to define a _query editor_ for Explore. Optionally, your plugin can also define a _start page_ for Explore.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/add-support-for-explore-queries.md:14]:
```suggestion
To support Explore, your plugin needs to do one or both of the following:
* Define a _query editor_ for Explore. 
* Define a _start page_ for Explore.
```

Comment (oddlittlebird) [reply docs/sources/developers/plugins/add-support-for-explore-queries.md:14]: What happens if they do both? Or is this an either/or?

File: docs/sources/developers/plugins/add-support-for-explore-queries.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+## Add a query editor
+
+The query editor for Explore is similar to the query editor for the data source itself. In fact, you'll probably reuse the same components for both query editors.
+
+To create a query editor for Explore:
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/add-support-for-explore-queries.md:20]: The colon at the end followed by a step that ends in a colon is weird. I suggest deleting this paragraph.

File: docs/sources/developers/plugins/add-support-for-explore-queries.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+This guide explains how to add support for [Explore]({{< relref "../../features/explore.md" >}}) to an existing data source.
+
+This guide assumes that you're already familiar with how to [Build a data source plugin]({{< relref "../../../../../tutorials/build-a-data-source-plugin.md" >}}).
+
+By adding support for Explore to your data source, users can make ad-hoc queries without the new of a dashboard. This is useful when users want to troubleshoot or to learn more about the data.
+
+To support Explore, your plugin needs to define a _query editor_ for Explore. Optionally, your plugin can also define a _start page_ for Explore.
+
+## Add a query editor
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/add-support-for-explore-queries.md:16]:
```suggestion
## Add a query editor for Explore
```

File: docs/sources/developers/plugins/add-support-for-explore-queries.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+         onRunQuery={props.onRunQuery}
+         onBlur={props.onBlur}
+         query={query.queryText || ''}
+         placeholder="Enter a query"
+       />
+     );
+   };
+   ```
+
+## Add a start page
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/add-support-for-explore-queries.md:91]:
```suggestion
## Add an Explore start page
```

File: docs/sources/developers/plugins/add-support-for-explore-queries.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+By adding an Explore start page for your plugin, you can for example create "cheat sheets" with commonly used queries. When the user clicks on one of the examples, it automatically updates the query editor, and runs the query. It's a great way to increase productivity for your users.
+
+To add a start page to your plugin:
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/add-support-for-explore-queries.md:95]:
```suggestion
```

**Review by oddlittlebird — APPROVED** [review]
Minor edits to address, then good to go.

File: docs/sources/developers/plugins/add-support-for-explore-queries.md (hunk 2 above)
Comment (marcusolsson) [reply docs/sources/developers/plugins/add-support-for-explore-queries.md:14]: The query editor is mandatory, the start page is optional. Your suggestion suggests that you could write only a start page, which is possible, but has little meaning without the query editor.

**Pull request merged**